### PR TITLE
Fix nonNegativeInteger - clean up config validators.

### PR DESCRIPTION
**Pull request by jfkhoury**
## Description

- The optional `idSyncContainer` config is causing an error because the `nonNegativeInteger` validator is was not allowing `undefined` values.
- Moved config validators from `core` to `Data Collector`.
- The `config` arg was never used in the validators.
- Came-cased `config-validators` to match the rest of the folders.

## Related Issue

https://jira.corp.adobe.com/browse/CORE-33402

## Types of changes

- [ X ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ X ] All tests pass and I've made any necessary test changes.
- [ X ] I have run the Sandbox successfully.

### CODE REVIEW

File: src/components/DataCollector/index.js [hunk 1]
```diff
@@ -73,4 +74,18 @@ const createDataCollector = ({ config }) => {
 
 createDataCollector.namespace = "DataCollector";
 createDataCollector.abbreviation = "DC";
+
```
Comment (liljenback) [inline src/components/DataCollector/index.js]: I thought we already had this before?

File: src/utils/configValidators/required.js [hunk 2]
```diff
@@ -10,7 +10,7 @@ OF ANY KIND, either express or implied. See the License for the specific languag
 governing permissions and limitations under the License.
 */
 
-export default (config, key, currentValue) => {
+export default (key, currentValue) => {
   let err = "";
   if (currentValue == null) {
```
Comment (Aaronius) [inline src/utils/configValidators/required.js]: I'd like to see this as `if (currentValue === undefined)`. I think validation should fail f the user passes in `null` for a required option.

File: src/utils/configValidators/nonNegativeInteger.js [hunk 3]
```diff
@@ -12,8 +12,9 @@ governing permissions and limitations under the License.
 
 import isInteger from "../isInteger";
 
-export default (config, key, currentValue) => {
-  return isInteger(currentValue) && currentValue >= 0
+export default (key, currentValue) => {
+  return currentValue === undefined ||
```
Comment (Aaronius) [inline src/utils/configValidators/nonNegativeInteger.js]: I like that you used `=== undefined` here.

File: src/utils/configValidators/eitherNilOrNonEmpty.js [hunk 4]
```diff
@@ -13,7 +13,7 @@ governing permissions and limitations under the License.
 import isNil from "../isNil";
 import isNonEmptyString from "../isNonEmptyString";
 
-export default (config, key, currentValue) => {
+export default (key, currentValue) => {
   if (isNil(currentValue)) {
```
Comment (Aaronius) [inline src/utils/configValidators/eitherNilOrNonEmpty.js]: I think this should be `if (currentValue === undefined){`. I think validation should fail if the user passes in `null`.

File: src/utils/configValidators/boolean.js [hunk 5]
```diff
@@ -12,7 +12,7 @@ governing permissions and limitations under the License.
 
 import isBoolean from "../isBoolean";
 
-export default (config, key, currentValue) => {
+export default (key, currentValue) => {
   return isBoolean(currentValue)
```
Comment (Aaronius) [inline src/utils/configValidators/boolean.js]: I believe this should return an empty string if the value is `undefined`. I'm not sure if you wanted to do that as part of this same PR.

File: src/utils/configValidators/eitherNilOrNonEmpty.js (hunk 4 above)
Comment (Aaronius) [reply src/utils/configValidators/eitherNilOrNonEmpty.js]: Oh wait.....nevermind on this one.

File: src/utils/configValidators/required.js (hunk 2 above)
Comment (Aaronius) [reply src/utils/configValidators/required.js]: OMG...I'm not thinking straight. Ignore this comment too.

File: src/components/DataCollector/index.js (hunk 1 above)
Comment (liljenback) [reply src/components/DataCollector/index.js]: Looks like it was removed by mistake, but this shiny new one looks better :)

**Pull request merged**